chain/client/src/view_client.rs

```diff
@@ -116,42 +121,69 @@ impl ViewClientActor {
     pub fn new(
         validator_account_id: Option<AccountId>,
         chain_genesis: &ChainGenesis,
-        runtime_adapter: Arc<dyn RuntimeAdapter>,
+        hot_runtime_adapter: Arc<dyn RuntimeAdapter>,
+        #[cfg(feature = "cold_store")] cold_runtime_adapter: Option<Arc<dyn RuntimeAdapter>>,
         network_adapter: Arc<dyn PeerManagerAdapter>,
         config: ClientConfig,
         request_manager: Arc<RwLock<ViewClientRequestManager>>,
         adv: crate::adversarial::Controls,
     ) -> Result<Self, Error> {
         // TODO: should we create shared ChainStore that is passed to both Client and ViewClient?
-        let chain = Chain::new_for_view_client(
-            runtime_adapter.clone(),
+        let hot_chain = Chain::new_for_view_client(
+            hot_runtime_adapter.clone(),
             chain_genesis,
             DoomslugThresholdMode::TwoThirds,
             !config.archive,
         )?;
+        #[cfg(feature = "cold_store")]
+        let cold_chain = cold_runtime_adapter
+            .as_ref()
+            .map(|runtime_adapter| {
+                Chain::new_for_view_client(
+                    runtime_adapter.clone(),
+                    chain_genesis,
+                    DoomslugThresholdMode::TwoThirds,
+                    !config.archive,
+                )
+            })
+            .transpose()?;
         Ok(ViewClientActor {
             adv,
             validator_account_id,
-            chain,
-            runtime_adapter,
+            hot_chain,
```

Comment:
Probably shouldn't be implemented in this PR. I think we should create a ViewClient class and probably moves the non-actor related code from ViewClientActor to ViewClient. This way, it is easier to create unit tests for ViewClient, which could be important for cold storage since now it adds many more functions in ViewClient.

Reply:
SG, though if this commit needs noticeable rewrites I’ll leave it to @posvyatokum to carry this over.

Reply:
Yes, totally. I think it might be better to do that in a separate PR, up to you!

Reply:
Ok, then I will take it over and close this PR